bridgesample/src/main/java/com/livefront/bridgesample/scenario/activity/NonBridgeLargeDataActivity.kt

```diff
@@ -39,17 +42,29 @@ class NonBridgeLargeDataActivity : NonBridgeBaseActivity() {
     }
 
     companion object {
-        private const val INFINTE_BACKSTACK_KEY = "infinite-backstack"
+        private const val ARGUMENTS_KEY = "arguments"
 
-        fun getInfiniteBackstack(
+        fun getArguments(
             activity: NonBridgeLargeDataActivity
-        ): Boolean = activity
+        ): NonBridgeLargeDataActivityArguments = activity
                 .intent
-                .getBooleanExtra(INFINTE_BACKSTACK_KEY, false)
+                .getParcelableExtra(ARGUMENTS_KEY)
 
-        fun getNavigationIntent(context: Context, infiniteBackstack: Boolean) = Intent(
+
+        fun getNavigationIntent(
+            context: Context,
+            arguments: NonBridgeLargeDataActivityArguments
+        ) = Intent(
                 context,
                 NonBridgeLargeDataActivity::class.java
-        ).apply { putExtra(INFINTE_BACKSTACK_KEY, infiniteBackstack) }
+        ).apply {
+            putExtra(ARGUMENTS_KEY, arguments)
+        }
     }
 }
+
+@Parcelize
+data class NonBridgeLargeDataActivityArguments(
+    val infiniteBackstack: Boolean = false
+) : Parcelable
+
```

Comment:
Probably don't need this extra line here.

Reply:
Done.